src/Symfony/Component/DependencyInjection/Compiler/ResolveParameterPlaceHoldersPass.php

```diff
@@ -65,10 +65,11 @@ protected function processValue($value, $isRoot = false)
             if (isset($changes['file'])) {
                 $value->setFile($this->bag->resolveValue($value->getFile()));
             }
-            $value->setProperties($this->bag->resolveValue($value->getProperties()));
-            $value->setMethodCalls($this->bag->resolveValue($value->getMethodCalls()));
+        }
+        if (is_array($value = parent::processValue($value, $isRoot)) && $value) {
```

Comment:
I would move the parent call outside the if, to be more clear (and remove potential mistakes where deciding to remove the handling of keys might also remove the parent call while it is always necessary)